pkg/providers/vsphere/virtualmachine/delete.go

```diff
@@ -1,22 +1,40 @@
-// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+// Copyright (c) 2022-2024 VMware, Inc. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
 package virtualmachine
 
 import (
+	"fmt"
+
 	"github.com/go-logr/logr"
 	"github.com/pkg/errors"
 	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/mo"
 	vimtypes "github.com/vmware/govmomi/vim25/types"
 
+	vmopv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha3"
 	pkgctx "github.com/vmware-tanzu/vm-operator/pkg/context"
+	"github.com/vmware-tanzu/vm-operator/pkg/providers/vsphere/constants"
 	vmutil "github.com/vmware-tanzu/vm-operator/pkg/util/vsphere/vm"
 )
 
 func DeleteVirtualMachine(
 	vmCtx pkgctx.VirtualMachineContext,
 	vcVM *object.VirtualMachine) error {
 
+	moVM := &mo.VirtualMachine{}
+	if err := vcVM.Properties(vmCtx, vcVM.Reference(), []string{"config.extraConfig"}, moVM); err != nil {
+		vmCtx.Logger.Error(err, "failed to get VM properties for DeleteVirtualMachine")
```

Comment:
```suggestion
		vmCtx.Logger.Error(err, "failed to fetch config.extraConfig properties of VM for DeleteVirtualMachine")
```

Basically, some way to call out "what property were you trying to fetch".  Could be helpful in debugging if this call fails.